### PR TITLE
Bap composite

### PR DESCRIPTION
**Pull request by VincentVerelst**
I made a minimally parametrized UDP for a BAP composite algorithm. Current parameters are temporal_extent, spatial_extent and bands (optional). 

Currently, max_cloud_cover, spatial_resolution, score weights and compositing window are fixed. 

During the process of adding this algortihm, I've gathered some comments which I'm happy to discuss!

### CODE REVIEW

File: algorithm_catalog/bap_composite_files/bap_composite_description.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Credits: 14 
+CPU usage: 7,214.715 cpu-seconds
+Wall time: 577 seconds
+Input Pixel 641.575 mega-pixel
+Max Executor Memory: 3.262 gb
+Memory usage: 25,839,702.134 mb-seconds
+Network Received: 71,733,702,645 b
+```
+
+The relative cost is 0.2 CDSE platform credits per km² for a 1 month input window.
```
Comment (jdries) [inline algorithm_catalog/bap_composite_files/bap_composite_description.md:46]: for 3 bands.

**Review by jdries — CHANGES_REQUESTED** [review]
a few smaller suggestions. If we have time, would be good to check out the artifacts.

File: algorithm_catalog/bap_composite.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      "rel": "service",
+      "type": "application/json",
+      "title": "CDSE openEO federation",
+      "href": "https://openeofed.dataspace.copernicus.eu"
+    },
+    {
+      "rel": "example",
+      "type": "application/json",
+      "title": "Example output",
+      "href": "https://s3.waw3-1.cloudferro.com/swift/v1/APEx-examples/bap_composite_denmark/collection.json"
```
[the remark this answers is not in the document]
Comment (VincentVerelst) [reply algorithm_catalog/bap_composite.json:96]: I checked the raw S2 data and it seems that in this month there just aren't any dates for which these pixels weren't clouded: https://openeo.dataspace.copernicus.eu/openeo/1.1/jobs/j-24101448a47641ec9e23fa41b37ad67c/results/NTc3ZjY1NDEtYzU2NS00Nzk4LWEyY2QtNWZjNDJkOTY0MDFi/f89125a1e86264618a79312746b96a24?expires=1729513023

[the remark this answers is not in the document]
Comment (VincentVerelst) [reply algorithm_catalog/bap_composite.json:96]: The STAC collection should now reference a better month with little to no artifacts

**Review by jdries — APPROVED** [review]
approved, I would merge it, allowing integration and further evaluation.

**Pull request merged**